### PR TITLE
fix `Couldn't find libgcc_s.so.1!`

### LINKED ISSUE #5732: repl `Couldn't find libgcc_s.so.1` on recent NixOS
```
» 2+2
Couldn't find libgcc_s.so.1!
You may need to install libgcc

We looked in the following directories:
/nix/store/1x4ijm9r1a88qk7zcmbbfza324gx1aac-glibc-2.37-8/lib
```

Unlike previous glibc releases the lib folder does not contain `libgcc_s.so.1`.
I'll investigate this.

### PR DESCRIPTION
**Pull request by Anton-4**
Closes #5732

### CODE REVIEW

File: crates/compiler/build/src/link.rs [hunk 1]
```diff
@@ -785,11 +786,25 @@ fn get_target_str(target: &Triple) -> &str {
...
+
+    paths
 }
 
-fn library_path<const N: usize>(segments: [&str; N]) -> Option<PathBuf> {
+fn nix_glibc_path_opt() -> Option<OsString> {
+    env::var_os("NIX_GLIBC_PATH")
+}
+
+fn build_path<const N: usize>(segments: [&str; N]) -> Option<PathBuf> {
```
Comment (Anton-4) [inline crates/compiler/build/src/link.rs:807]: build_path is a better name.

File: crates/compiler/build/src/link.rs [hunk 2]
```diff
@@ -812,22 +827,21 @@ fn library_path<const N: usize>(segments: [&str; N]) -> Option<PathBuf> {
...
+            let mut path_cl = path.clone();
+            path_cl.push(lib_filename);
+            path_cl
         })
         .find(|path| path.exists())
 }
 
+fn strs_to_path(strs: &[&str]) -> PathBuf {
+    strs.iter().collect()
+}
```
Comment (Anton-4) [inline crates/compiler/build/src/link.rs:843]: This makes repeated conversion a lot cleaner.

File: crates/compiler/test_gen/benches/list_map.rs [hunk 3]
```diff
@@ -62,8 +62,13 @@ fn roc_function<'a, 'b>(
...
 
     let context = inkwell::context::Context::create();
-    let (main_fn_name, errors, lib) =
-        helpers::llvm::helper(arena, config, source, arena.alloc(context));
+    let (main_fn_name, errors, lib) = helpers::llvm::helper(
+        arena,
+        config,
+        source,
+        arena.alloc(context),
+        roc_load::FunctionKind::LambdaSet,
```
Comment (Anton-4) [inline crates/compiler/test_gen/benches/list_map.rs:70]: Fixes an error for these benchmarks.

File: flake.nix [hunk 4]
```diff
@@ -125,9 +125,15 @@
               [ ]);
 
           LLVM_SYS_130_PREFIX = "${llvmPkgs.llvm.dev}";
+
           # nix does not store libs in /usr/lib or /lib
+          # for libgcc_s.so.1
+          NIX_LIBGCC_S_PATH =
+            if pkgs.stdenv.isLinux then "${pkgs.stdenv.cc.cc.lib}/lib" else "";
```
Comment (Anton-4) [inline flake.nix:132]: `libgcc_s.so.1` used to be present in the NIX_GLIBC_PATH but this is no longer the case with recent releases.

File: flake.nix [hunk 5]
```diff
@@ -136,7 +142,7 @@
             1; # to run the editor with NVIDIA's closed source drivers
           
           shellHook = ''
-            source <(echo "${aliases}")
+            ${aliases}
```
Comment (Anton-4) [inline flake.nix:145]: The aliases did not work before :p

**Review by rtfeldman — APPROVED** [review]
Awesome, thank you @Anton-4! 🎉

**Review by brian-carroll — APPROVED** [review]

**Pull request merged**